src/com/thizthizzydizzy/treefeller/compat/MMOCoreCompat.java

```diff
@@ -236,6 +247,152 @@ public String writeToConfig(HashMap<String, Double> value){
+            return new ItemBuilder(Material.GOLDEN_PICKAXE);
+        }
+        @Override
+        public void openGlobalModifyMenu(MenuGlobalConfiguration parent){
+            parent.open(new MenuModifyStringIntegerMap(parent, parent.plugin, parent.player, name, 0, Integer.MAX_VALUE, false, false, globalValue, ( value) -> {
+                globalValue = value;
+            }));
+        }
+        @Override
+        public void openToolModifyMenu(MenuToolConfiguration parent, Tool tool){
+            parent.open(new MenuModifyStringIntegerMap(parent, parent.plugin, parent.player, name, 0, Integer.MAX_VALUE, true, false, toolValues.get(tool), (value) -> {
+                if(value==null)toolValues.remove(tool);
+                else toolValues.put(tool, value);
+            }));
+        }
+        @Override
+        public void openTreeModifyMenu(MenuTreeConfiguration parent, Tree tree){
+            parent.open(new MenuModifyStringIntegerMap(parent, parent.plugin, parent.player, name, 0, Integer.MAX_VALUE, true, false, treeValues.get(tree), (value) -> {
+                if(value==null)treeValues.remove(tree);
+                else treeValues.put(tree, value);
+            }));
+        }
+        @Override
+        public String writeToConfig(HashMap<String, Integer> value){
+            if(value==null)return "";
+            String s = "{";
+            String str = "";
+            for(String st : value.keySet()){
+                str+=", "+st+": "+value.get(st);
+            }
+            if(!str.isEmpty())s+=str.substring(2);
+            return s+"}";
+        }
+        @Override
+        protected DebugResult doCheck( TreeFeller plugin, Tool tool, Tree tree, Block block, Player player, ItemStack axe ){
+            //System.out.println( "[SOP] hello from mmocore compat OPTION" );
+    
+    
+            HashMap<String, Integer> lvlReqs = this.get( tool, tree );
+    
+            if( lvlReqs == null ){
+                //System.out.println( "[SOP] lvlreq null, returning" );
+                return null;
+            }
+            net.Indyuce.mmocore.api.player.PlayerData data = net.Indyuce.mmocore.api.player.PlayerData.get( player );
+    
+            boolean allowFell = true;
+            for( String profession : lvlReqs.keySet( ) ){
+                int lvlReq = lvlReqs.get( profession );
+                int playerLevel;
+    
+                if( profession.equals( "global" ) ){
+                    playerLevel = data.getLevel( );
+                } else {
+                    playerLevel = data.getCollectionSkills( ).getLevel( profession );
+                }
+    
+                if( lvlReq > playerLevel ){
+                    //System.out.println( "[SOP] You need at least level " + lvlReq + " " + profession + " to chop down this tree" );
+                    player.sendMessage( "You need at least level " + lvlReq + " " + profession + " to chop down this tree" );

```

Comment:
Messages should be sent through the Message system (pass the arguments through the DebugResult) to allow for customization/localization (Additionally, only debug messages should be set by default)